src/transport/SessionHandle.h

```diff
@@ -85,6 +91,7 @@ class SessionHandle
     NodeId mPeerNodeId;
     Optional<uint16_t> mLocalSessionId;
     Optional<uint16_t> mPeerSessionId;
+    Optional<GroupId> mGroupId;
```

Comment:
I'm not familiar with group messages. The `SessionHandle` is acting as a reference to the actual session object, and it should only contains the essential assets in order to locate a session. (I'll replace such assets with a raw pointer to the session object in the future).

I'm not sure if `GroupId` is being used as the key to locate the group session. If you are confident about it, you can use it here, and please make sure, this field is not exposed by `SessionHandle`.

Reply:
The idea of the groupcast session context in the spec is that it is ephemeral and need only exist on the stack so long as a single groupcast message is being processed, and wouldn't be needed longer than this as no multicast replies or exchanges are possible.

The context is supposed to hold the necessary information to understand the origin of a groupcast message in order to process groupcast application messages and, when necessary, initiate MCSP with the sender (where MCSP does require tracking a context with the peer over an exchange, but this is a different session context).

The unique identifier for the groupcast session context is <fabric index, group ID, source node ID>.  Source IP address, port, and operational group key must additionally be known to initiate MCSP.

I'm not entirely sure what is the best way to handle this.  The session handle is a way to provide a lookup in the session manager and to match to application context.  In the case of groupcast, there will never be an ongoing session and there will never be any preexisting application context.

Reply:
The groupcast session handle isn't created until after our branch for MCSP handling, so doesn't need to contain any information to support MCSP.

All we need is to make sure that the <fabric index, group ID, source node ID> source can be known through the entire unsolicited message handler path.  mGroupId by itself in the session handle is ambiguous.  And fabric index isn't directly inferrable from the payload header.

The simplest approach with existing code is to add all of <fabric index, group ID, source node ID> to the session handle.  That's only 11 bytes.  This doesn't seem too bad.

Reply:
@msandstedt all of <fabric index, group ID, source node ID> is already present in this implementation. However, @kghost 's refactor should preserve this.

Reply:
Ah, I see.  Then this makes sense.

Thanks.